modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/MatchOnlyTextFieldMapper.java

```diff
@@ -325,22 +333,16 @@ private IOFunction<LeafReaderContext, CheckedIntFunction<List<Object>, IOExcepti
             final KeywordFieldMapper.KeywordFieldType keywordDelegate
         ) {
             if (keywordDelegate.ignoreAbove().valuesPotentiallyIgnored()) {
-                // because we don't know whether the delegate field will be ignored during parsing, we must also check the current field
```

Comment:
I've realized there was some redundancy here. First of all, we never need to check `delegateFieldFallbackName` because if the keyword delegate ignores a value, then the parent field will be responsible for storing it. In other words, we need to check `fallbackName`. Second, we never need to check `fieldName` because match_only_text fields are not stored and don't have doc_values. 

With those two redundancies removed, we have two scenarios left:
- the value is stored in a fallback field of this match_only_text field (when there is no keyword delegate or the keyword delegate ignores the values)
- the value is stored by the keyword delegate field